### ISSUE TITLE
Cross Network Use of Accounts Should Return Warning Message

### ISSUE THREAD

**Issue by andersonmmi** [MetaMask/metamask-extension#3429]
*  I expected that MetaMask would warn me against using Truffle TestRPC accounts on the Main Net, or at least silo accounts between networks. I also expect metamask would return an error if I tried to purchase ETH for a TestRPC account.
  *  MetaMask allowed me to import the private keys from Truffle TestRPC and interact with the Main Net.  I was able to link through the buy button and fund this account with 0.001 Ether from Coinbase.  That ETH was captured within 3 blocks.
  *  Browser: Chrome
  *  OS: MacOS 10.13.3

  
![screen shot 2018-03-04 at 3 38 55 pm](https://user-images.githubusercontent.com/16314437/36951967-6538c730-1fcf-11e8-8884-eb1c1bf28582.png)

**Comment by Zanibas** [comment]
Although I understand your concerns, would you have any ideas by which we could even detect this kind of behavior? If it were as simple as detecting if a user has used a non-mainnet testnet within a single session, then we could set some booleans in the background to trigger a warning message.

However, it seems in this case that you've imported a private key from Truffle. There isn't a very good way for us to detect if this is a "test address" on the various chains out there.

However again, this was a test address that is basically becoming a "whoops" ground for several people--would it help you if we just blacklist this specific address and warn people that this is the truffle address as a stopgap measure?

**Comment by andersonmmi** [comment]
You know how imported addresses are labelled as loose?  We could just label these testRPC/ganache addresses with "test".  Seems like a logic test against the array of known test addresses could check this on the front end and it would be very light.  I can code it up an PR it if that would help?

**Comment by Zanibas** [comment]
Yes please, go ahead and start up the PR and we can continue discussions there--thank you for such due diligence on writing up that blog post!
